### DISCUSSION TITLE
500 Error with Next-intl Middleware on Vercel Deployment: EDGE_FUNCTION_INVOCATION_FAILED

### DISCUSSION THREAD

**Discussion by Zakisb** [vercel/next.js#74609] [Help] [answered]
### Link to the code that reproduces this issue

https://github.com/Zakisb/aimplifyy

### To Reproduce

Encountering a server-side error when deploying a Next.js 14 application with next-intl middleware to Vercel. The deployment fails with a 500 status code, specifically citing EDGE_FUNCTION_INVOCATION_FAILED as the reason. Additionally, a ReferenceError: __dirname is not defined occurs within the ua-parser-js module, as well as in the user-agent and middleware files of next-intl. This error is not present during local development and only manifests upon deployment to Vercel.

Note that previously i worked with next-intl and deployed without a problem. i don't know if it's something to do with the latest next 14 version or next-intl

### Current vs. Expected behavior

I expected the web app to be deployed. Currently it's crashing 
![Screenshot 2023-11-07 at 3 56 36 PM](https://github.com/vercel/next.js/assets/28612240/28ccd4b3-e2c5-4286-8193-2ca90ae32756)
![Screenshot 2023-11-07 at 3 56 49 PM](https://github.com/vercel/next.js/assets/28612240/36a5b3c1-3774-41c4-9bfe-0190e0ecd571)

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:34 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T8103
Binaries:
  Node: 18.18.0
  npm: 9.8.1
  Yarn: 1.22.11
  pnpm: N/A
Relevant Packages:
  next: 14.0.1
  eslint-config-next: 14.0.0
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Middleware / Edge (API routes, runtime)

### Additional context

Works fine locally.

**Comment by intelcoder** [comment]
I am having same issue as well

**Comment by rosso007** [comment]
same issue here

_Edit by maintainers: Comment was **automatically** minimized because it was considered unhelpful. (If you think this was by mistake, let us know). Please only comment if it adds context to the issue. If you want to express that you have the same problem, use the upvote 👍 on the issue description or subscribe to the issue for updates. Thanks!_

**Comment by samcx** [comment]
@Zakisb Thank you for sharing!

I currently receive an error when attempting to build your <picture data-single-emoji=":repro:" title=":repro:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/repro/5cea53b5532825b4.png" alt=":repro:" align="absmiddle"></picture> locally (`next build`).

```
> Export encountered errors on following paths:
	/[locale]/page: /en
	/[locale]/page: /fr
 ELIFECYCLE  Command failed with exit code 1.
 ```

**Comment by MichaelWalker-git** [comment]
For me, I ended up having to upgrade from the free tier to the paid tier on the number of Vercel functions.

**Comment by samcx** [comment]
@MichaelWalker-git Are you referring to this Serverless Function limit → https://vercel.com/docs/functions/serverless-functions/runtimes#functions-created-per-deployment (separate behavior from above, which is referring to an [Edge Function](https://vercel.com/docs/functions/edge-functions) error).

**Comment by norio-club** [comment]
Same issue!

_Edit by maintainers: Comment was **automatically** minimized because it was considered unhelpful. (If you think this was by mistake, let us know). Please only comment if it adds context to the issue. If you want to express that you have the same problem, use the upvote 👍 on the issue description or subscribe to the issue for updates. Thanks!_

**Comment by RobinDeNeef** [comment, marked as answer]
I had the same issue. I set up different projects to try to pinpoint the problem. 

I noticed my faulty project had the wrong "Build & Development Settings" in the project settings on Vercel.

It was set to "other" instead of "Next.js", which caused it to build differently with `npm run vercel-build`. 

![image](https://github.com/vercel/next.js/assets/16370409/182cabc6-a5de-4195-a8ee-d5fba690347b)

Can you verify if these settings are set to "NextJS"? Switching it to "NextJS" did the trick for me.

**Reply by Pandahoccode** [reply]
> You are such a genius ! Thank you I'm stuck with it for 2 days OMG that was awsome. Thanks a lot !

**Comment by rosso007** [comment]
@RobinDeNeef This did the trick for me as well :)

**Comment by raytawil-masterfulai** [comment]
> I had the same issue. I set up different projects to try to pinpoint the problem.
> 
> I noticed my faulty project had the wrong "Build & Development Settings" in the project settings on Vercel.
> 
> It was set to "other" instead of "Next.js", which caused it to build differently with `npm run vercel-build`.
> 
> ![image](https://private-user-images.githubusercontent.com/16370409/295073239-182cabc6-a5de-4195-a8ee-d5fba690347b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1MDgxMjYsIm5iZiI6MTcwNTUwNzgyNiwicGF0aCI6Ii8xNjM3MDQwOS8yOTUwNzMyMzktMTgyY2FiYzYtYTVkZS00MTk1LWE4ZWUtZDVmYmE2OTAzNDdiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE3VDE2MTAyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2MGU3NGJjNTEwMTc4MmM2NWFhMzczNjYxNTkzOGY4ZTlmZmJiM2ZmM2Q0Y2NhY2M5YmVkYjhiNDgxYTFmNzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tQhoWl3dnGgRxVNSxYCyyG17K0QRqMc_-UD0LUzM2pg)
> 
> Can you verify if these settings are set to "NextJS"? Switching it to "NextJS" did the trick for me.

That fixed it for me as well. Thanks man!

**Comment by DoyeDesigns** [comment]
This fixed it for me. Thanks man

**Comment by dlmarshall3** [comment]
This thread is proof to start with Google before spending [redacted amount of time] arguing with an LLM.
